### ISSUE TITLE
Choropleth example

### ISSUE THREAD

**Issue by adamConnerSax** [DougBurke/hvega#94]
I just got a chance to look at the tutorial and it's awesome! SO helpful to have examples. 

It's missing a chloropleth and I struggled a bit to get one working.  Two issues, mainly:  I was not so familiar with the "lookup" syntax which you need to join data to the geofeatures and it's sort of hard to figure out what field you need to do your lookup on.  I was trying to match "states" but ultimately needed to use "properties.name".  Which all makes sense now that I have done it.  

If I translated one of the standard chloropleth examples, probably https://vega.github.io/vega-lite/docs/geoshape.html, would you accept a PR to the tutorial or, if I post the hvega code here, would you add it?

**Comment by DougBurke** [comment]
The lookup stuff isn't obvious to me either. I know it has changed a bit in Vega-Lite v4 (or at least the elm module - see https://github.com/gicentre/elm-vegalite/issues/14 for an example of my lack of understanding).

I'm happy to see updates to the tutorial or just placed here. I think I have an un-pushed branch somewhere where I was trying to update one of the tutorial examples, but don't let that stop you as I don't have time at the moment for hvega, thanks to a flock of deadlines screaming by.

**Comment by adamConnerSax** [comment]
Yep.  I'll get to it in the next week or two and make a PR.  Then you can incorporate as you see fit.  I'll paste the code here as well so that people looking can have a better shot at finding the answer.  

Thanks again for all the work.  The new version is great!

**Comment by DougBurke** [comment]
Looking forward to it.

**Comment by jwoLondon** [comment]
(apologies for being the boring spelling pedant, but I mention this now in case it otherwise ends up in the documentation following the PR...)

It's "choropleth" not "chloropleth"

**Comment by adamConnerSax** [comment]
Yeah. Realized that...

Thanks!

Adam

**Comment by adamConnerSax** [comment]
I am still planning to do this!  Just got busy.  Should be able to get to it soon.  I've made many choropleths since so I think the examples will be better!

**Comment by DougBurke** [comment]
Given my lack of progress on `hvega` I don't think I can complain ;-)

**Comment by adamConnerSax** [comment]
:)
It's actually sort of tricky because there are two ways to do the lookup:
1.  Do a lookup in your data and attach your value (or your row) to the geo feature.  But that won't work for something faceted where you need more than one row per geo feature.  For that you need
2. To do a lookup in the geo data and attach the geo feature to your data set.  Which would also work for the first case.  So maybe it should be the canonical way.  
Anyway, I'll try to add an example of each.  Someday.  Soon.

**Comment by adamConnerSax** [comment]
Making progress!  Found my way around the tutorial setup and made a simple example.  You can see here: https://github.com/adamConnerSax/hvega/tree/AddChoroplethExamples. 
However, for the more complex one, a faceted thing, there is something (```repeat``` or ```resolve```?) which I think needs vega-embed v4 and the current setup does not make that easy to do since it's embedded in the toHtmlWith function.  I'm happy to do the work to make this more flexible, but I'm not sure how you'd like it done.  Pointers?

**Comment by adamConnerSax** [comment]
Ah!  I see you've been working on v4.  That'd solve my issue, I think.  I also made a PR to add some versioning control.  But you might prefer to just upgrade to v4 and skip it.  Whenever you merge either your v4 stuff or my versioning stuff (or both!) to master, I'll finish this and submit a PR.

**Comment by DougBurke** [comment]
Apropos of nothing, one fun thing I've noticed is that the output (of Vega Embed) partly depends on what version of Vega Embed you are using (e.g. v3 or v4), and not the version of the Vega-Lite spec [in that if I use Vega Embed 4 to view a v3 schema file then I *think* I no longer see the tooltips by default behavior, which you do see if you viewed the same file with Vega Embed 3. Or at least I'm fairly sure I've seen this].

**Comment by adamConnerSax** [comment]
Huh.  Never did much comparing!
Do you have thoughts about how I ought to move forward on these examples?  You should also feel free to just add them (from the linked repo above) as you do your v4 work.

**Comment by DougBurke** [comment]
Sorry - I have my head down with #106 to the exclusion of all else.

**Comment by DougBurke** [comment]
https://github.com/DougBurke/hvega/blob/choropleth/hvega/src/Graphics/Vega/Tutorials/VegaLite.hs#L5384 is my attempt to rebase this onto the current msater branch (it doesn't include the changes in #108 but that shouldn't affect these plots). The `FromGeo` plot now creates "valid" JSON, in that it passes the 4.0.2 spec, but isn't doing the right thing - it doesn't seem to be properly handling the repeated field, which I assume is something to do with how lookup/repeat handling is handled by hvega.

https://vega.github.io/editor/#/url/vega-lite/N4KABGBEBOCmAOsCGAXSAuMwYHsDuGYA2pPDvAK4A2qAljgHaQA0UsDA5rQ7LNAM4soACwrRotAMZIeggLoBfZuCj9EkwqAgRIKaDP4AzHNAC2hIiu1Yr1qEkGZIHWDha3rkQ9BznMWu08AE1QkTQ9AqDEqQkhhFBR4fnQAehT9PAA6LhRRACMKfj5JRhR2FEyS0xSAN1gOJFr6xtMHMugUoJxJfk7QlMKAWgBGAAZTTIArfkZ3SLsvE1a0fy9kFDFYWP4UVFhBVl0AT0RYlHIcadmFCOslW+1IAGtYI9jaIMgH+-nIKhwcE8KPB3p8IjdrHJlJ5WtAnrEXDh+MIkKdoY8QrtwoFINFYvFEsk0hlsrRchQCkVoCUGGVaZVfE0GkyWm0+J1ur1MY0yJQaCh6AwAPrsLg8Pj8IWicRSGT7Sr8GpfO7onR4D65QgAZlGo1VUDgMyodU0kH40ioW1WJX+0HeDCCCHYjtpkAUP0e8B8k1gkgFs1WKBOVvsVDyEoAqvwwh6dOwSkFuBxsQsbSZTYZaLAqJ9VnBEKhYj4CEooEHTk4AI4UGQC3YCk2xx7I1Eh7CZ7O5qCIoTHCvd1xXJgQ7Qjj2QAAk5uEsFa+ISSVSLNJ5LymXoKWns8adQagyoZNgtQALFMZkx0ZAaZnk-4bgogA

**Comment by adamConnerSax** [comment]
I’ll take a look tonight or tomorrow!

**Comment by adamConnerSax** [comment]
Okay!  The fix is simple.  Move the 
``` 
resolve
       . resolution (RScale [(ChColor, Independent)])
       $ []
```
to the arguments to ```toVegaLite``` and out of the individual spec.  It wasn't a lookup problem.  There are just a lot more people than engineers or hurricanes. :)

It's probably easiest for you to just do it, but if you want me to submit a PR, I can do that tomorrow.  No time right now to get a fork at the right spot...

**Comment by DougBurke** [comment]
I can add it to my rebased version of your branch - may take a day or two.

**Comment by DougBurke** [comment]
You can see what I've done in #110

**Comment by adamConnerSax** [comment]
Looks good!  Does it work for you when you run the tutorial?  Do you want/need anything else from me?

Thanks again for all the work to catch up with vega-lite!

**Comment by DougBurke** [comment]
Closed by #110
